CHANGELOG.md

```diff
@@ -23,6 +23,20 @@
   **Access to container from multiple threads should be handled by clients** from now on.
 * All `resolve` methods now can throw.
 
+	### Note on migration from 3.x to 4.0.0:
+	* Errors
+
+	In 4.0.0 each `resolve` method can throw `DefinitionNotFound(DefinitionKey)` error, so you need to call it using `try!` or `try?`, or catch the error if it's appropriate for your case. See [#15](https://github.com/AliSoftware/Dip/issues/15) for rationale of this change.
+	
+	* Thread safety
+
+	In 4.0.0 `DependencyContainer` drops any gauaranty of thread safety. From now on code that uses Dip must ensure that it's methods are called from a single thread. For example if you have registered type as a singleton and later two threads try to resolve it at the same time you can have two different instances of type instead of one as expected.
```

Comment:
+ `guarantee`
+ Maybe add rationale on why we had to do this as feels like an unwanted regression otherwise.